### PR TITLE
For #26690 - Dismiss search dialog when opening recent synced tab dropdown menu

### LINKED ISSUE #26690: [Bug]: "Remove" button from the last synced tab from jump back in doesn't work when the cursor in the search bar is active
### Steps to reproduce

1. On a mobile device login in with a Firefox account.
2. On a second device login with the same account and open at least 1 tab.
3. Perform a sync.
4. On the mobile homescreen go to the jump back in section and long tap on the last synced tab while the cursor from the search bar is active.
5. Tap on "Remove".

### Expected behaviour

The tab is removed. The remove button is no longer displayed.

### Actual behaviour

The "Remove" button doesn't work.

### Device name

Oppo Find X5

### Android version

Android 12

### Firefox release type

Firefox Nightly

### Firefox version

106.0a1

### Device logs

_No response_

### Additional information

https://user-images.githubusercontent.com/48829350/187222042-e7ff3c5a-f780-40a9-8f7b-892017d202c0.mp4

The button works as expected when the cursor is not active.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-21413)

### PR DESCRIPTION
**Pull request by Alexandru2909**
Added call to dismiss search dialog on long clicking the recent synced tabs. Allowing the user to interact with the `Remove` dropdown.

https://user-images.githubusercontent.com/35462038/188153391-b4ee767e-d6d3-477d-9132-cf59897ff212.mp4

### Pull Request checklist

- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### QA

- [x] **QA Needed**

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-debug` task.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
Fixes #26690

### CODE REVIEW

**Review by gabrielluong — COMMENTED** [review]
The current bug also happens for the recent tab. Can we also fix that there?

**Comment by Alexandru2909** [comment]
> The current bug also happens for the recent tab. Can we also fix that there?

I opened https://github.com/mozilla-mobile/fenix/issues/26790 for this.

File: app/src/main/java/org/mozilla/fenix/home/recentsyncedtabs/interactor/RecentSyncedTabInteractor.kt [hunk 1]
```diff
@@ -17,6 +17,11 @@ interface RecentSyncedTabInteractor {
      */
     fun onRecentSyncedTabClicked(tab: RecentSyncedTab)
 
+    /**
+     * Called when opening the dropdown menu on a recent tab by long press.
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/home/recentsyncedtabs/interactor/RecentSyncedTabInteractor.kt:21]:
```suggestion
     * Called when opening the dropdown menu on a recent synced tab by long press.
```

**Review by gabrielluong — APPROVED** [review]

**Comment by Mugurell** [comment]
Seems like this can be landed after fixing the conflicts with `main` - incorporating the trailing commas. I can help with that.

**Pull request merged**